src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,392 @@
+                " `XLAFSDPStrategy` does not use the `CheckpointIO`."
+            )
+        # broadcast the path from rank 0 to ensure all the states are saved in a common path
+        path = self.broadcast(path)
+        if Path(path).is_dir() and os.listdir(path):
+            raise FileExistsError(f"The checkpoint directory already exists and is not empty: {path}")
+        from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
+
+        modules = [module for module in state.values() if isinstance(module, XLAFSDP)]
+        if len(modules) == 0:
+            raise ValueError(
+                "Could not find a XLAFSDP model in the provided checkpoint state. Please provide the model as"
+                " part of the state like so: `save_checkpoint(..., state={'model': model, ...})`. Make sure"
+                " you set up the model (and optimizers if any) through the strategy before saving the checkpoint."
+            )
+        if len(modules) > 1:
+            raise ValueError(
+                "Found multiple XLAFSDP modules in the given state. Saving checkpoints with FSDP is"
+                " currently limited to a single model per checkpoint. To save multiple models, call the"
+                " save method for each model separately with a different path."
+            )
+        rank = self.local_rank
+        world_size = self.world_size
+        import torch_xla.core.xla_model as xm
+
+        # ensure model parameters are updated
+        xm.mark_step()
+
+        converted_state: Dict[str, Any] = {}
+        for key, obj in state.items():
+            # convert the state
+            if isinstance(obj, Module) and isinstance(obj, XLAFSDP):
+                converted = obj.state_dict()
+                # add shard_metadata to state
+                converted_state["shard_metadata"] = obj.get_shard_metadata()
+            elif isinstance(obj, Optimizer):
+                converted = obj.state_dict()
+            else:
+                converted = obj
+            _apply_filter(key, filter or {}, converted, converted_state)
+
+        self.checkpoint_io.save_checkpoint(
+            converted_state,
+            os.path.join(path, f"checkpoint_rank-{rank:08d}-of-{world_size:08d}.pth"),
+            storage_options=storage_options,
+        )
+
+        if self._state_dict_type == "full":
+            from torch_xla.distributed.fsdp import consolidate_sharded_model_checkpoints
+
+            if self.is_global_zero:
+                consolidate_sharded_model_checkpoints(
+                    ckpt_prefix=os.path.join(path, "checkpoint"), ckpt_suffix="_rank-*-of-*.pth"
+                )
+            self.barrier("ckpt_consolidation")
+            self.checkpoint_io.remove_checkpoint(
+                os.path.join(path, f"checkpoint_rank-{rank:08d}-of-{world_size:08d}.pth")
+            )
+
+    def remove_checkpoint(self, folderpath: _PATH) -> None:
```

Comment:
It looks like this method is not used anywhere. The calls above use the checkpoint_io. 
I suggest we either call this method or we remove it.

Reply:
I can add `remove_checkpoint` to the integration testing. However, it is worth pointing out when we are deleting shards in `save_checkpoint()` (when `self._state_dict_type == "full"`), we cannot reference `remove_checkpoint` since due to the fact that `self._state_dict_type == "full"`, we would delete the full checkpoint and not the shards. That is why we call `self.checkpoint_io.remove_checkpoint()` directly.

Reply:
Added `remove_checkpoint()` this to the integration tests

Reply:
@gkroiz It is unclear to me where this method is used. If it is not used in the code base anywhere, why do we need to define and test it?

Reply:
Hm... this is a good point. I originally added this since the `xla` strategy also has `remove_checkpoint()`. IMO, I think we should add this to `remove_checkpoint()` to fabric since it could be helpful and I don't see any harm. The user could type `fabric.remove_checkpoint()` which would call the strategy's `remove_checkpoint()`. Up to you guys whether this is a good addition, otherwise we can remove `remove_checkpoint()` defined within the strategies.

Reply:
You are right, this would be a valid use case, for example someone implementing a checkpoint callback or similar. I don't have a strong opinion on this, not a blocker at all!

Reply:
in that case, we can leave in for the time being and can remove in a later PR if needed. Will resolve this conversation.

Reply:
Since https://github.com/Lightning-AI/lightning/pull/18167 is removing it. You can remove it with this PR.

Since it's unused internally and undocumented externally, it's the safest option. We can always re-introduce it if users need it with proper documentation and design

Reply:
Sounds good, just removed